From 464002dfcccfb9bec31abc410592fc9543b0b44f Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Mon, 21 Aug 2017 00:46:49 +0100
Subject: [PATCH 10/35] Settings: Allow devices to provide remote gesture
 preferences

 * Not all devices can use common gestures as implemented in LineageParts

Change-Id: Iae4ca9fc686e3dc82411524688841211e0cceb7b
---
 res/values/cm_strings.xml | 4 ++++
 res/xml/gestures.xml      | 9 +++++++++
 2 files changed, 13 insertions(+)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index fa615f818b..de1a870f7f 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -37,4 +37,8 @@
     <!-- Wake on plug -->
     <string name="wake_when_plugged_or_unplugged_title">Wake on plug</string>
     <string name="wake_when_plugged_or_unplugged_summary">Turn the screen on when connecting or disconnecting a power source</string>
+
+    <!-- Touchscreen gesture settings -->
+    <string name="touchscreen_gesture_settings_title">Touchscreen gestures</string>
+    <string name="touchscreen_gesture_settings_summary">Perform various touchscreen gestures for quick actions</string>
 </resources>
diff --git a/res/xml/gestures.xml b/res/xml/gestures.xml
index ef5b733e51..e86a63a8d8 100644
--- a/res/xml/gestures.xml
+++ b/res/xml/gestures.xml
@@ -26,6 +26,15 @@
         android:key="touchscreen_gesture_settings"
         lineage:requiresFeature="lineagehardware:FEATURE_TOUCHSCREEN_GESTURES" />
 
+    <lineageos.preference.RemotePreference
+            android:key="device_touchscreen_gesture_settings"
+            android:title="@string/touchscreen_gesture_settings_title"
+            android:summary="@string/touchscreen_gesture_settings_summary"
+            lineage:replacesKey="touchscreen_gesture_settings"
+            lineage:requiresFeature="lineagehardware:FEATURE_TOUCHSCREEN_GESTURES">
+        <intent android:action="org.lineageos.settings.device.GESTURE_SETTINGS" />
+    </lineageos.preference.RemotePreference>
+
     <Preference
         android:key="gesture_assist_input_summary"
         android:title="@string/assist_gesture_title"
-- 
2.17.1

